### ISSUE TITLE
[3.4] Next-gen setcontent parser exceptions

### ISSUE THREAD

**Issue by GawainLynch** [bolt/bolt#6902]
With a :koala: in hand and `config.yml` set to:
```
compatibility:
    setcontent_legacy: false
```
Things go :boom:, probably due to my helpfulness getting the branch across.
```
SyntaxErrorException in AbstractSQLiteDriver.php line 78:
An exception occurred while executing 'SELECT _pages.*, 
GROUP_CONCAT(incomingrelation.id) as _incomingrelation_id, 
GROUP_CONCAT(incomingrelation.from_id) as _incomingrelation_fromid, 
GROUP_CONCAT(incomingrelation.from_contenttype) as _incomingrelation_fromcontenttype, 
GROUP_CONCAT(groups.sortorder) as _groups_sortorder, 
GROUP_CONCAT(groups.id) as groups_id, 
GROUP_CONCAT(groups.slug) as _groups_slug, 
GROUP_CONCAT(groups.name) as _groups_name, 
GROUP_CONCAT(groups.taxonomytype) as _groups_taxonomytype 
FROM bolt_pages _pages 
LEFT JOIN bolt_relations incomingrelation 
ON _pages.id = incomingrelation.to_id 
AND incomingrelation.to_contenttype='pages' 
LEFT JOIN bolt_taxonomy groups 
ON _pages.id = groups.content_id 
AND groups.contenttype='pages' 
AND groups.taxonomytype='groups' 
GROUP BY _pages.id, _pages.id 
ORDER BY datechanged DESC ASC LIMIT 5':

SQLSTATE[HY000]: General error: 1 near "ASC": syntax error
```

Could you be so kind as to have a look @rossriley

**Comment by rossriley** [comment]
I think I've already fixed this somewhere... although I might not have PRd it yet... let me do some excavation and I'll update.

**Comment by rossriley** [comment]
Yes, I remember what it is now. This line is the culprit:

https://github.com/bolt/bolt/blob/3.4/src/Controller/Async/General.php#L502

The syntax in the parser needs `-datechanged` rather than `datechanged DESC`

I'm assuming though that the current syntax works ok on the legacy system in which case although it might not be documented anywhere we should perhaps adjust the new parser to support both forms.

Alternatively we can just change it here to `-datechanged`. Thoughts?

**Comment by GawainLynch** [comment]
Ha! This one again! How do I keep turning it up :rofl: 

I don't think it always works as expected, and the last time I managed to find one of these nobody else can remember seeing it. It certainly isn't documented and we checked right back to v1 docs to be sure.

Let's go with updating this (I have the file open next to me so I can quickly PR it)

**Comment by GawainLynch** [comment]
Fixed in #6924 … Thanks mate!
